src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -195,7 +206,12 @@ public void testWithCacheWithNoViolation() throws Exception {
         final File file = temporaryFolder.newFile("file.java");
         final List<String> lines = new ArrayList<>();
         lines.add(" class a {} ");
-        treeWalker.processFiltered(file, lines);
+        try {
+            treeWalker.processFiltered(file, lines);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception is not expected");
```

Comment:
IMO, this shouldn't be the expected way to build tests.
Exceptions should be thrown up past the method to the runner so it can display the full stacktrace to the user for debugging if it is not expected.
We shouldn't catch and fail with a simple message.
We should be asserting some actual result in this test.